src/serialization/sb3.js

```diff
@@ -297,16 +297,34 @@ const getExtensionIdForOpcode = function (opcode) {
  * compressed primitives and the list of all extension IDs present
  * in the serialized blocks.
  */
-const serializeBlocks = function (blocks) {
+const serializeBlocks = function (blocks, extensionManager) {
     const obj = Object.create(null);
     const extensionIDs = new Set();
     for (const blockID in blocks) {
         if (!Object.prototype.hasOwnProperty.call(blocks, blockID)) continue;
-        obj[blockID] = serializeBlock(blocks[blockID], blocks);
         const extensionID = getExtensionIdForOpcode(blocks[blockID].opcode);
         if (extensionID) {
             extensionIDs.add(extensionID);
+            const instance = extensionManager.getExtensionInstance(extensionID);
+            var extensionBlocks = instance.info.blocks;
+            extensionBlocks = extensionBlocks.reduce((acc, tempBlock) => {
+                acc[tempBlock.opcode] = tempBlock;
+                return acc;
+            }, {});
+            var blockInfoIndex = blocks[blockID].opcode;
+            if (!Object.keys(primitiveOpcodeInfoMap).includes(blocks[blockID].opcode)) {
+                blockInfoIndex = blocks[blockID].opcode.replace(`${blocks[blockID].opcode.split("_")[0]}_`, "");
+            }
+            const regex = /_v\d+/g; // Matches _v followed by one or more digits
+            blockInfoIndex = blockInfoIndex.replace(regex, ""); // Replaces all matches with an empty string
+            const menuRegex = /menu_\d+/;
+            if (!menuRegex.test(blockInfoIndex)) {
```

Comment:
Interesting! so we don't apply version to menus, makes sense -- I do wonder what happens when a project is saved with a menu value that disappears in later versions. Is that something that's handled?